group-coordinator/src/main/java/org/apache/kafka/coordinator/group/RecordHelpers.java

```diff
@@ -357,6 +366,94 @@ public static Record newCurrentAssignmentTombstoneRecord(
         );
     }
 
+    /**
+     * Creates a GroupMetadata record.
+     *
+     * @param group              The generic group.
+     * @param memberAssignments  The assignment by member id.
+     * @param metadataVersion    The metadata version.
+     * @return The record.
+     */
+    public static Record newGroupMetadataRecord(
+        GenericGroup group,
+        Map<String, byte[]> memberAssignments,
+        MetadataVersion metadataVersion
+    ) {
+        short version;
+        if (metadataVersion.isLessThan(IBP_0_10_1_IV0)) {
+            version = (short) 0;
+        } else if (metadataVersion.isLessThan(IBP_2_1_IV0)) {
+            version = (short) 1;
+        } else if (metadataVersion.isLessThan(IBP_2_3_IV0)) {
+            version = (short) 2;
+        } else {
+            version = (short) 3;
+        }
```

Comment:
While we are here, let's push this logic into `MetadataVersion`. We did this for others.